docs/developer/validation-rules/resource-model-validation.md

```diff
@@ -0,0 +1,38 @@
+# Identifying resource model definitions in a specification
+## Description
+A model definition is considered as a resource model definition if:
+
+- For each model, check if that model is returned by a PUT operation as a part of 200 or 201 response codes.
+- If the model has an 'allOf' Resource at any level [name of the root model is 'Resource'] AND the model is returned by a GET operation in 200 response.
+- If the model has 'x-ms-azure-resource': true or if any model has an 'x-ms-azure-resource' set to true in its model hierarchy and is not named either of 'Resource', 'TrackedResource' or 'ProxyResource', consider this as a resource
+
+## Types of resource models
+1. TrackedResource: A resource is considered a tracked resource if a model hierarchy has a property named 'location' which is marked as required.
+2. ProxyResource: Proxy resources are all resources without a required 'location' property in their model hierarchy
+
+## Validation rules for resource models
+
+### PutGetPatchResponseInvalid 
+For a Given Resource, GET/PUT/PATCH MUST return the same `Resource` Model. The GET/PUT/PATCH operations under a path should reference the same response model schema.
+### How to fix it
+Ensure GET/PUT/PATCH operations under a path reference the same resource model in their response model schemas.
+
+### PutRequestResponseValidation
+A PUT operation request body schema should be the same as its 200 response schema, to allow reusing the same entity between GET and PUT. If the schema of the PUT request body is a superset of the GET response body, there must be a PATCH operation to make the resource updatable.
+### How to fix it
+Ensure the PUT operation request and response schemas are the same. If they are not the same, ensure there is a PATCH operation to make the resource updatable.
+
+### ResourceIsMsResourceValidation
+Every `Resource` Model MUST be tagged with `x-ms-azure-resource`. This will indicate Autorest to make the Resource model inherit from the Resource definition in the client runtime.
```

Comment:
Is the second sentence true in every language? I'm not sure. We could say "In some languages, this will...."